components/brave_wallet/browser/eth_allowance_manager.cc

```diff
@@ -0,0 +1,409 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/eth_allowance_manager.h"
+
+#include <algorithm>
+#include <utility>
+
+#include "brave/components/brave_wallet/browser/blockchain_registry.h"
+#include "brave/components/brave_wallet/browser/brave_wallet_utils.h"
+#include "brave/components/brave_wallet/browser/json_rpc_service.h"
+#include "brave/components/brave_wallet/browser/keyring_service.h"
+#include "brave/components/brave_wallet/browser/pref_names.h"
+#include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
+#include "brave/components/brave_wallet/common/hash_utils.h"
+#include "brave/components/brave_wallet/common/hex_utils.h"
+#include "components/prefs/pref_service.h"
+#include "components/prefs/scoped_user_pref_update.h"
+
+namespace {
+constexpr int kTopicElementsCountToCheck = 3;
+constexpr char kLastBlockNumber[] = "last_block_number";
+constexpr char kAllowanceFound[] = "allowances_found";
+constexpr char kApproverAddress[] = "approver_address";
+constexpr char kContractAddress[] = "contract_address";
+constexpr char kSpenderAddress[] = "spender_address";
+constexpr char kAmount[] = "amount";
+
+constexpr char kAddress[] = "address";
+constexpr char kTopics[] = "topics";
+constexpr char kFromBlock[] = "fromBlock";
+constexpr char kToBlock[] = "toBlock";
+constexpr char kEarliestBlock[] = "earliest";
+constexpr char kLatestBlock[] = "latest";
+
+constexpr char kApprovalTopicHash[] = "Approval(address,address,uint256)";
```

Comment:
nit: this is not a hash, just a function signature

Reply:
fixed